profiles/util.py

```diff
@@ -29,15 +35,93 @@ def split_name(name):
         return names[0], " ".join(names[1:])
 
 
-def profile_image_upload_uri(_, filename):
+def _generate_upload_to_uri(suffix=""):
     """
-    Helper to format the uri for the profile image upload
+    Returns a function to specify the upload directory and filename, via upload_to on an ImageField
+
+    Args:
+        suffix (str):
+            A suffix for the filename
+    Returns:
+        function:
+            A function to use with upload_to to specify an upload directory and filename
     """
-    name, ext = path.splitext(filename)
-    timestamp = datetime.now(pytz.utc).replace(microsecond=0)
 
-    return "profile/{name}-{timestamp}{ext}".format(
-        name=name[:IMAGE_FILENAME_MAX_LENGTH],
-        timestamp=timestamp.strftime("%Y-%m-%dT%H%M%S-%z"),
-        ext=ext,
-    )
+    def _upload_to(_, filename):
+        """Function passed to upload_to on an ImageField"""
+        name, ext = path.splitext(filename)
+        timestamp = datetime.now(pytz.utc).replace(microsecond=0)
+        path_format = "profile/{name}-{timestamp}{suffix}{ext}"
+
+        path_without_name = path_format.format(
+            timestamp=timestamp.strftime("%Y-%m-%dT%H%M%S-%z"),
+            suffix=suffix,
+            ext=ext,
+            name='',
+        )
+        if len(path_without_name) >= IMAGE_PATH_MAX_LENGTH:
+            raise Exception("path is longer than max length even without name: {}".format(path_without_name))
```

Comment:
If it fails here then `suffix` is too large. Since the end user should not have control over `suffix` I think an exception is the way to go here. Otherwise we would need to cut off `name` or `timestamp`